.github/workflows/bump-version.yml

```diff
@@ -0,0 +1,244 @@
+            if [ "${bump_major}" != "${current_major}" ]; then
+                echo "ERROR: ${{ github.event.inputs.part }} release attempted, but ${{ github.event.inputs.new_version }} major version not equal to ${current_tag}."
+                exit 1
+            fi
+            # Minor versions should be equal
+            if [ "${bump_minor}" != "${current_minor}" ]; then
+                echo "ERROR: ${{ github.event.inputs.part }} release attempted, but ${{ github.event.inputs.new_version }} minor version not equal to ${current_tag}."
+                exit 1
+            fi
+            if [ "${bump_patch}" != "$((${current_patch} + 1))" ]; then
+                if ! ([ "${bump_patch}" == "${current_patch}" ] && [ ${{ github.event.inputs.release_candidate }} == 'true' ]); then
+                    echo "ERROR: ${{ github.event.inputs.part }} release candidate release attempted, but ${{ github.event.inputs.new_version }} is more than 1 ${{ github.event.inputs.part }} version greater then ${current_tag}."
+                    exit 1
+                fi
+            fi
+        fi
+
+        echo "  ...version bump validated!"
+        if [ ${{ github.event.inputs.release_candidate }} == 'true' ]; then
+            echo "* Bumping version ${current_tag} to ${{ github.event.inputs.part }} version release candidate ${{ github.event.inputs.new_version }}"
+        else
+            echo "* Bumping version ${current_tag} to ${{ github.event.inputs.part }} version ${{ github.event.inputs.new_version }}"
+        fi
+
+        echo ::set-output name=old_tag::v${current_tag}
+
+    - name: Set up Python 3.10
+      if: success()
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+
+    - name: Install Python dependencies
+      run: |
+        python -m pip install --upgrade pip setuptools wheel
+        python -m pip install tbump
+        python -m pip list
+
+    - name: Setup Git user to push new tag
+      run: |
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+
+    - name: Bump version and push to GitHub
+      if: >-
+        github.event_name == 'workflow_dispatch'
+        && (
+          github.event.sender.login == 'lukasheinrich' ||
+          github.event.sender.login == 'matthewfeickert' ||
+          github.event.sender.login == 'kratsg'
+        )
+      shell: bash
+      run: |
+        tbump --non-interactive --no-push ${{ github.event.inputs.new_version }}
+
+    - name: Update the Git tag annotation
+      if: ${{ github.event.inputs.dry_run }} == 'false'
+      shell: bash
+      run: |
+        OLD_TAG=${{ steps.script.outputs.old_tag }}
+        git tag -n99 --list "${OLD_TAG}"
+
+        NEW_TAG=v${{ github.event.inputs.new_version }}
+        git tag -n99 --list "${NEW_TAG}"
+
+        CHANGES=$(git log --pretty=format:'%s' "${OLD_TAG}"..HEAD --regexp-ignore-case --extended-regexp --grep='^([a-z]*?):')
```

Comment:
This section is mostly taken from the tag workflow. Reminder that we filter on PRs that have a semantic PR title so we only grab `<semantic PR type>: PR title`. (Well we don't check that it is a semantic PR type, but the CI does for us).